spynnaker/pyNN/models/neural_projections/projection_machine_edge.py

```diff
@@ -56,8 +56,38 @@ def filter_edge(self, graph_mapper):
                 pre_hi = graph_mapper.get_slice(self.pre_vertex).hi_atom
                 post_lo = graph_mapper.get_slice(self.post_vertex).lo_atom
                 post_hi = graph_mapper.get_slice(self.post_vertex).hi_atom
-                if pre_hi < post_lo or pre_lo > post_hi:
-                    n_filtered += 1
+                # Filter edge if both are views and outside limits
+                if ((synapse_info.prepop_is_view) and
+                        (synapse_info.postpop_is_view)):
+                    prepop_lo = synapse_info.pre_population._indexes[0]
+                    prepop_hi = synapse_info.pre_population._indexes[-1]
+                    postpop_lo = synapse_info.post_population._indexes[0]
+                    postpop_hi = synapse_info.post_population._indexes[-1]
+                    if ((prepop_hi < pre_lo or prepop_lo > pre_hi) or
+                            (postpop_hi < post_lo or postpop_lo > post_hi)):
+                        n_filtered += 1
+                # Filter edge if pre-pop is outside limit and post_lo is bigger
+                # than n_pre_neurons
+                elif synapse_info.prepop_is_view:
+                    prepop_lo = synapse_info.pre_population._indexes[0]
+                    prepop_hi = synapse_info.pre_population._indexes[-1]
+                    prepop_size = synapse_info.n_pre_neurons
+                    if ((prepop_hi < pre_lo or prepop_lo > pre_hi) or
+                            (prepop_size < post_lo)):
```

Comment:
As above pre vs. post

Reply:
This is a bit more confusing; the first part is the same as the above case, but it's also necessary to check the size of the pre-population against the low value of the post slice (the post-pop in this case is not a view, so the OneToOne will start from index 0 and continue for as long as there are values from the pre-pop View available to it).

Reply:
I think this suffers from the same issue (again not wrong) e.g.:
```
a: |--|01|23|
b: |01|23|
```
Again a[2:3] -> b[2:3] should be filtered but I don't think it is.